src/Symfony/Component/Security/Core/Security.php

```diff
@@ -57,4 +58,23 @@ public function getToken(): ?TokenInterface
     {
         return $this->container->get('security.token_storage')->getToken();
     }
+
+    /**
+     * Logout the current user automatically. Dispatch the logout event.
+     */
+    public function logout(): void
+    {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $logoutEvent = new LogoutEvent($request, $this->container->get('security.token_storage')->getToken());
+
+        $firewallName = $this
+            ->container
+            ->get('security.firewall.map')
+            ->getFirewallConfig($request)
+            ->getName()
+        ;
+
+        $this->container->get('security.firewall.event_dispatcher_map')->getEventDispatcher($firewallName)->dispatch($logoutEvent);
+        $this->container->get('security.token_storage')->setToken();
```

Comment:
As [suggested by Jordi](https://github.com/symfony/symfony/pull/41406#discussion_r741705363), I would prefer if we change this method's return type to `?Response` and return `$logoutEvent->getResponse()`.

Reply:
Thanks, fixed.